src/fuzz/spanner_emulator_ddl_statement_proto_to_string_test.cc

```diff
@@ -0,0 +1,328 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <string>
+
+#include "src/fuzz/protobufs/create_table.pb.h"
+#include "src/fuzz/protobufs/spanner_ddl.pb.h"
+#include "src/fuzz/protobufs/utils/spanner_emulator_ddl_statement_proto_to_string.h"
+#include "gtest/gtest.h"
+#include <google/protobuf/repeated_field.h>
+
+#include "absl/strings/substitute.h"
+#include "absl/strings/str_cat.h"
+
+using spanner_ddl::SpannerFuzzingStatements;
+using spanner_ddl::SpannerDDLStatement;
+using spanner_ddl::CreateTable;
+using spanner_ddl::Column;
+using spanner_ddl::ColumnDataInfo;
+
+//TODO: add more to this test as more APIs are added
+TEST(DDLStatementProtoToString, DDLStatementToString) {
+    SpannerFuzzingStatements statements;
+    SpannerDDLStatement* statement1 = statements.add_statements();
+
+    CreateTable* createTable = statement1->mutable_createtable();
+    createTable->set_tablename("testTable");
+
+    Column* column1 = createTable->add_primarykeys();
+    column1->set_columnname("testColumn1");
+    
+    ColumnDataInfo* columnDataInfo1 = column1->mutable_columndatainfo();
+    columnDataInfo1->set_scalartype(ColumnDataInfo::STRING);
+    columnDataInfo1->set_length(200);
+    columnDataInfo1->set_lengthtype(ColumnDataInfo::UNBOUND);
+    columnDataInfo1->set_isarray(false);
+    column1->set_isnotnull(true);
+    column1->set_allowcommittimestamp(true);
+    column1->set_orientation(Column::ASC);
+
+    Column* column2 = createTable->add_primarykeys();
+    column2->set_columnname("testColumn2");
+    
+    ColumnDataInfo* columnDataInfo2 = column2->mutable_columndatainfo();
+    columnDataInfo2->set_scalartype(ColumnDataInfo::BOOL);
+    columnDataInfo2->set_length(2);
+    columnDataInfo2->set_lengthtype(ColumnDataInfo::MAX);
+    columnDataInfo2->set_isarray(true);
+    column2->set_isnotnull(false);
+    column2->set_allowcommittimestamp(false);
+    column2->set_orientation(Column::DESC);
+
+    Column* column3 = createTable->add_nonprimarykeys();
+    column3->set_columnname("testColumn3");
+
+    ColumnDataInfo* columnDataInfo3 = column3->mutable_columndatainfo();
+    columnDataInfo3->set_scalartype(ColumnDataInfo::TIMESTAMP);
+    columnDataInfo3->set_length(-100);
+    columnDataInfo3->set_lengthtype(ColumnDataInfo::BOUND);
+    columnDataInfo3->set_isarray(false);
+    column3->set_isnotnull(false);
+    column3->set_allowcommittimestamp(true);
+
+    EXPECT_EQ(createTable->primarykeys().size(), 2);
+    EXPECT_EQ(createTable->nonprimarykeys().size(), 1);
+
+    EXPECT_EQ(toString(*statement1), 
+        absl::StrCat("CREATE TABLE testTable ([ { testColumn1 STRING( 200 )",
```

Comment:
This doesn't look like correct DDL.  I believe it should be:

CREATE TABLE testTable ( 
   testColumn1 STRING( 200 ) NOT NULL OPTIONS (allow_commit_timestamp=true),
   testColumn2 ARRAY< BOOL >  OPTIONS ( allow_commit_timestamp = null ), 
   testColumn3 TIMESTAMP OPTIONS (allow_commit_timestamp = true )
) PRIMARY KEY (testColumn1 ASC, testColumn2 DESC)

Also:  consider using a multi-line string over absl::StrCat here, as it's easier to read.

That looks something like: "sdl(*string contents here*)sdl"